### PR TITLE
Masun dev

### PR DESCRIPTION
**Pull request by maggiemsft**
This change implements string trim char and try to solve issue discussed in https://github.com/dotnet/coreclr/issues/1826
Appreciate comments and review
This is also the bootcamp assignment
cc: @ellismg @sneshaf @anniepoh

### CODE REVIEW

File: src/mscorlib/src/System/String.cs [hunk 1]
```diff
@@ -1287,17 +1287,39 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
...
         }
-    
-    
-        // Removes a set of characters from the end of this string.
+       
+        // Removes single character from the Start or/and end of this string.
         [Pure]
-        public String Trim(params char[] trimChars) {
-            if (null==trimChars || trimChars.Length == 0) {
+        public String Trim(char c) {
```
Comment (justinvp) [inline src/mscorlib/src/System/String.cs]: Should the parameter be named `trimChar` to be consistent with the existing `trimChars`?

**Comment by justinvp** [comment]
@ellismg, if this PR can go in coreclr *master*, can we also go ahead and merge #895 in coreclr *master*? Then submit a separate PR against corefx *future* to actually expose the APIs in the reference assembly?

File: src/mscorlib/src/System/String.cs (hunk 1 above)
Comment (maggiemsft) [inline src/mscorlib/src/System/String.cs]: Agreed. Should I submit a new change and PR for it?

Comment (ellismg) [inline src/mscorlib/src/System/String.cs]: You can just create a new commit (or amend the existing one) and then push the new changes to GitHub.  You don't need to create a new PR.

**Comment by jkotas** [comment]
Dup of #2070

**Pull request closed without merging**